packages/playwright/src/reporters/html.ts

```diff
@@ -511,7 +511,12 @@ class HtmlBuilder {
       startTime: result.startTime.toISOString(),
       retry: result.retry,
       steps: dedupeSteps(result.steps).map(s => this._createTestStep(s, result)),
-      errors: formatResultFailure(internalScreen, test, result, '').map(error => error.message),
+      errors: formatResultFailure(internalScreen, test, result, '').map(error => {
+        return {
+          message: error.message,
+          codeframe: error.location ? createErrorCodeframe(error.message, error.location) : undefined
```

Comment:
Don't we already have a codeframe inside `error.message`, or even `error.snippet`? Let's reuse that.